lib/model/model.go

```diff
@@ -1233,12 +1234,16 @@ func (m *model) ccHandleFoldersLocked(folders []protocol.Folder, deviceCfg confi
 }
 
 // requires fmut read-lock
-func (m *model) ccCheckEncryptionLocked(fcfg config.FolderConfiguration, folderDevice config.FolderDeviceConfiguration, ccDevice, ccDeviceUs protocol.Device, hasDevice, hasUs bool) error {
+func (m *model) ccCheckEncryptionLocked(fcfg config.FolderConfiguration, folderDevice config.FolderDeviceConfiguration, ccDevice, ccDeviceUs protocol.Device, hasDevice, hasUs, deviceUntrusted bool) error {
 	hasTokenDev := hasDevice && len(ccDevice.EncPwToken) > 0
 	hasTokenUs := hasUs && len(ccDeviceUs.EncPwToken) > 0
 	isEncDev := folderDevice.EncryptionPassword != ""
 	isEncUs := fcfg.Type == config.FolderTypeReceiveEncrypted
 
+	if !isEncDev && !isEncUs && deviceUntrusted {
```

Comment:
The variable names are horrible (and were horrible from the previous PRs), but shouldn't this be:

`cfg.Type  == config.FolderTypeReceiveEncrypted && untrusted && folderDevice.EncryptionPassword == ""`

which is

`!isEncDev && isEncUs && deviceUntrusted`

Reply:
I don't think so: When we receive encrypted content (`cfg.Type == config.FolderTypeReceiveEncrypted`), we don't need to encrypt to an untrusted devices (well actually we can't, it doesn't make sense and it's an error condition further down). This says: If we don't have encrypted data locally, nor encrypt our plain data for the given device and that device is untrusted, that's bad.

Just because they were in a previous PR doesn't mean I can't change them anymore - shoot away with better ones.

Reply:
But isn't in that case we're just proxying what we got?
Why do we need a password in that case?

Reply:
We don't need a pw then: When we are proxying encrypted stuff `cfg.Type == config.FolderTypeReceiveEncrypted` is true and thus this error doesn't get emitted.

Reply:
Right, get it. 

I think we should fix the Enc Pw stuff to more readable names, and us and dev shoudl be local and remote.